packages/client/src/components/Conclusion/NewCheckerModal.test.tsx

```diff
@@ -20,71 +22,85 @@ import NewCheckerModal from "./NewCheckerModal";
-      queryByTestId(`radio-checker-${customTopic.slug}`) as HTMLElement
-    );
-
-    // Because this event is not async, it should call `NO_CHOICE_HAS_BEEN_MADE`
-    fireEvent.click(queryByTestId(MODAL_CONFIRM_BUTTON) as HTMLElement);
-
-    expect(mockMatomoTrackEvent).toHaveBeenCalledTimes(3);
-    expect(mockMatomoTrackEvent).toBeCalledWith({
-      action: actions.START_ANOTHER_CHECK,
-      name: `${eventNames.DO_ANOTHER_CHECK} - ${eventNames.NO_CHOICE_HAS_BEEN_MADE}`,
+    act(() => {
+      fireEvent.click(queryByTestId(MODAL_OPEN_BUTTON) as HTMLElement);
     });
-  });
+    // Modal = open
 
-  // Go to new topic
-  await act(async () => {
-    fireEvent.click(queryByTestId(MODAL_CONFIRM_BUTTON) as HTMLElement);
+    expect(queryByTestId(MODAL)).toBeInTheDocument();
+    expect(queryByText(customTopic.name)).toBeInTheDocument();
+    expect(queryByText("Annuleer")).toBeInTheDocument();
 
-    expect(mockMatomoTrackEvent).toHaveBeenCalledTimes(4);
+    // Make sure Matomo Analytics work
+    // TODO: temp disabled this matamo tracking test
+    expect(mockMatomoTrackEvent).toHaveBeenCalledTimes(1);
     expect(mockMatomoTrackEvent).toBeCalledWith({
       action: actions.OPEN_MODAL,
       name: eventNames.OPEN_MODAL_DO_ANOTHER_CHECK,
     });
-
-    // expect(window.location.href).toContain(customTopic.slug);
   });
 
-  // @TODO: This test could be extended to verify that the SessionData has been updated
+  it("should open a new topic", async () => {
+    const { queryByTestId } = render(<NewCheckerModal />);
+
+    act(() => {
+      fireEvent.click(queryByTestId(MODAL_OPEN_BUTTON) as HTMLElement);
+    });
+    // Modal = open
+
+    // Expect the "same address" components to be rendered with this `useTopicData` mock
+    expect(queryByTestId(NEW_CHECKER_MODAL_SAME_ADDRESS)).toBeInTheDocument();
+
+    act(() => {
+      // Select 'Yes' (only to mock the 'Yes')
+      fireEvent.click(queryByTestId(RADIO_ADDRESS_1) as HTMLElement);
+      // Select 'No' (because we don't send address data now)
+      fireEvent.click(queryByTestId(RADIO_ADDRESS_2) as HTMLElement);
+      // Click the new topic to open
+      fireEvent.click(
+        queryByTestId(`radio-checker-${customTopic.slug}`) as HTMLElement
+      );
+
+      // Because this event is not async, it should call `NO_CHOICE_HAS_BEEN_MADE`
+      fireEvent.click(queryByTestId(MODAL_CONFIRM_BUTTON) as HTMLElement);
+
+      expect(mockMatomoTrackEvent).toBeCalledWith({
+        action: actions.START_ANOTHER_CHECK,
+        name: `${eventNames.DO_ANOTHER_CHECK} - ${eventNames.NO_CHOICE_HAS_BEEN_MADE}`,
+      });
+    });
+
+    // Go to new topic
+    await act(async () => {
+      fireEvent.click(queryByTestId(MODAL_CONFIRM_BUTTON) as HTMLElement);
+
+      expect(mockMatomoTrackEvent).toBeCalledWith({
+        action: actions.OPEN_MODAL,
+        name: eventNames.OPEN_MODAL_DO_ANOTHER_CHECK,
+      });
+
+      // expect(window.location.href).toContain(customTopic.slug);
```

Comment:
What's up with this comment?

Reply:
It was already there. I think it was disabled because it doesn't work anymore with the new hooks functionality. I removed it for now.